### ISSUE TITLE
Compile failed on Ubuntu in version 0.80.

### ISSUE THREAD

**Issue by soloist-v** [twistedfall/opencv-rust#455]
I was able to successfully compile version 0.80 on Windows, but it fails when I try to compile it on Ubuntu. However, if I downgrade to version 0.77, I am able to compile it on Ubuntu without any issues.
1. Operating system：
```
Ubuntu 18.04.6 LTS (GNU/Linux 5.4.0-146-generic x86_64)
```
2. The way to installed OpenCV: 
```
[dependencies]
opencv = "0.80.0"
```
3. OpenCV version
```
0.80.0
```
4. rustc version (`rustc --version`)
```
rustc 1.70.0-nightly (f63ccaf25 2023-03-06)
```
5. Full output of `cargo build -vv`
https://gist.github.com/soloist-v/7d4f7093911761b12f85119acd1f3e04

**Comment by soloist-v** [comment]
Caused by:
  process didn't exit successfully: `CARGO=/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=opencv CARGO_MANIFEST_DIR=/home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.80.0 CARGO_PKG_AUTHORS='Pro <twisted.fall@gmail.com>:Mathieu Poumeyrol <kali@zoy.org>' CARGO_PKG_DESCRIPTION='Rust bindings for OpenCV' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=opencv CARGO_PKG_REPOSITORY='https://github.com/twistedfall/opencv-rust' CARGO_PKG_RUST_VERSION=1.59 CARGO_PKG_VERSION=0.80.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=80 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/home/ubuntu/Rust/hikvision_sdk/target/debug/deps:/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib:/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib' OUT_DIR=/home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-6ddc7cd013a105f9/out rustc --crate-name opencv --edition=2021 /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.80.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=142 --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="alphamat"' --cfg 'feature="aruco"' --cfg 'feature="aruco_detector"' --cfg 'feature="barcode"' --cfg 'feature="bgsegm"' --cfg 'feature="bioinspired"' --cfg 'feature="calib3d"' --cfg 'feature="ccalib"' --cfg 'feature="cudaarithm"' --cfg 'feature="cudabgsegm"' --cfg 'feature="cudacodec"' --cfg 'feature="cudafeatures2d"' --cfg 'feature="cudafilters"' --cfg 'feature="cudaimgproc"' --cfg 'feature="cudaobjdetect"' --cfg 'feature="cudaoptflow"' --cfg 'feature="cudastereo"' --cfg 'feature="cudawarping"' --cfg 'feature="cvv"' --cfg 'feature="default"' --cfg 'feature="dnn"' --cfg 'feature="dnn_superres"' --cfg 'feature="dpm"' --cfg 'feature="face"' --cfg 'feature="features2d"' --cfg 'feature="flann"' --cfg 'feature="freetype"' --cfg 'feature="fuzzy"' --cfg 'feature="gapi"' --cfg 'feature="hdf"' --cfg 'feature="hfs"' --cfg 'feature="highgui"' --cfg 'feature="img_hash"' --cfg 'feature="imgcodecs"' --cfg 'feature="imgproc"' --cfg 'feature="intensity_transform"' --cfg 'feature="line_descriptor"' --cfg 'feature="mcc"' --cfg 'feature="ml"' --cfg 'feature="objdetect"' --cfg 'feature="optflow"' --cfg 'feature="ovis"' --cfg 'feature="phase_unwrapping"' --cfg 'feature="photo"' --cfg 'feature="plot"' --cfg 'feature="quality"' --cfg 'feature="rapid"' --cfg 'feature="rgbd"' --cfg 'feature="saliency"' --cfg 'feature="sfm"' --cfg 'feature="shape"' --cfg 'feature="stereo"' --cfg 'feature="stitching"' --cfg 'feature="structured_light"' --cfg 'feature="superres"' --cfg 'feature="surface_matching"' --cfg 'feature="text"' --cfg 'feature="tracking"' --cfg 'feature="video"' --cfg 'feature="videoio"' --cfg 'feature="videostab"' --cfg 'feature="viz"' --cfg 'feature="wechat_qrcode"' --cfg 'feature="xfeatures2d"' --cfg 'feature="ximgproc"' --cfg 'feature="xobjdetect"' --cfg 'feature="xphoto"' -C metadata=e8dfcf29ea4386d4 -C extra-filename=-e8dfcf29ea4386d4 --out-dir /home/ubuntu/Rust/hikvision_sdk/target/debug/deps -L dependency=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps --extern libc=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/liblibc-9fba6e94e0c98d18.rmeta --extern num_traits=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/libnum_traits-04d49bac869cb071.rmeta --extern once_cell=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/libonce_cell-a6df48fd9f0cab30.rmeta --cap-lints warn -L native=/home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-6ddc7cd013a105f9/out -L /usr/lib/x86_64-linux-gnu -l static=ocvrs -l stdc++ -l opencv_shape -l opencv_stitching -l opencv_superres -l opencv_videostab -l opencv_aruco -l opencv_bgsegm -l opencv_bioinspired -l opencv_ccalib -l opencv_datasets -l opencv_dpm -l opencv_face -l opencv_freetype -l opencv_fuzzy -l opencv_hdf -l opencv_line_descriptor -l opencv_optflow -l opencv_video -l opencv_plot -l opencv_reg -l opencv_saliency -l opencv_stereo -l opencv_structured_light -l opencv_phase_unwrapping -l opencv_rgbd -l opencv_viz -l opencv_surface_matching -l opencv_text -l opencv_ximgproc -l opencv_calib3d -l opencv_features2d -l opencv_flann -l opencv_xobjdetect -l opencv_objdetect -l opencv_ml -l opencv_xphoto -l opencv_highgui -l opencv_videoio -l opencv_imgcodecs -l opencv_photo -l opencv_imgproc -l opencv_core --cfg ocvrs_opencv_branch_32 --cfg ocvrs_has_module_aruco --cfg ocvrs_has_module_bgsegm --cfg ocvrs_has_module_bioinspired --cfg ocvrs_has_module_calib3d --cfg ocvrs_has_module_ccalib --cfg ocvrs_has_module_core --cfg ocvrs_has_module_dpm --cfg ocvrs_has_module_face --cfg ocvrs_has_module_features2d --cfg ocvrs_has_module_flann --cfg ocvrs_has_module_freetype --cfg ocvrs_has_module_fuzzy --cfg ocvrs_has_module_hdf --cfg ocvrs_has_module_highgui --cfg ocvrs_has_module_imgcodecs --cfg ocvrs_has_module_imgproc --cfg ocvrs_has_module_line_descriptor --cfg ocvrs_has_module_ml --cfg ocvrs_has_module_objdetect --cfg ocvrs_has_module_optflow --cfg ocvrs_has_module_phase_unwrapping --cfg ocvrs_has_module_photo --cfg ocvrs_has_module_plot --cfg ocvrs_has_module_rgbd --cfg ocvrs_has_module_saliency --cfg ocvrs_has_module_shape --cfg ocvrs_has_module_stereo --cfg ocvrs_has_module_stitching --cfg ocvrs_has_module_structured_light --cfg ocvrs_has_module_superres --cfg ocvrs_has_module_surface_matching --cfg ocvrs_has_module_text --cfg ocvrs_has_module_video --cfg ocvrs_has_module_videoio --cfg ocvrs_has_module_videostab --cfg ocvrs_has_module_viz --cfg ocvrs_has_module_ximgproc --cfg ocvrs_has_module_xobjdetect --cfg ocvrs_has_module_xphoto` (exit status: 1)

**Comment by soloist-v** [comment]
(.env) ubuntu@ubuntu-X11DPi-N-T:~/Rust/hikvision_sdk$ cargo tree -d
libc v0.2.142
├── io-lifetimes v1.0.10
│   ├── is-terminal v0.4.7
│   │   └── flexi_logger v0.25.3
│   │       └── hikvision_sdk v0.1.0 (/home/ubuntu/Rust/hikvision_sdk)
│   └── rustix v0.37.14
│       └── is-terminal v0.4.7 (*)
├── numpy v0.18.0
│   └── hikvision_sdk v0.1.0 (/home/ubuntu/Rust/hikvision_sdk)
├── opencv v0.80.0
│   └── hikvision_sdk v0.1.0 (/home/ubuntu/Rust/hikvision_sdk)
├── parking_lot_core v0.9.7
│   └── parking_lot v0.12.1
│       └── pyo3 v0.18.3
│           ├── hikvision_sdk v0.1.0 (/home/ubuntu/Rust/hikvision_sdk)
│           └── numpy v0.18.0 (*)
├── pyo3 v0.18.3 (*)
├── pyo3-ffi v0.18.3
│   └── pyo3 v0.18.3 (*)
└── rustix v0.37.14 (*)

libc v0.2.142
├── clang v2.0.0
│   └── opencv-binding-generator v0.60.0
│       [build-dependencies]
│       └── opencv v0.80.0 (*)
│   [build-dependencies]
│   └── opencv v0.80.0 (*)
├── clang-sys v1.6.1
│   ├── bindgen v0.65.1
│   │   [build-dependencies]
│   │   └── hikvision_sdk v0.1.0 (/home/ubuntu/Rust/hikvision_sdk)
│   ├── clang v2.0.0 (*)
│   └── opencv-binding-generator v0.60.0 (*)
├── jobserver v0.1.26
│   └── cc v1.0.79
│       [build-dependencies]
│       └── opencv v0.80.0 (*)
│   [build-dependencies]
│   └── opencv v0.80.0 (*)
└── which v4.4.0
    └── bindgen v0.65.1 (*)

log v0.4.17
└── bindgen v0.65.1 (*)

log v0.4.17
├── flexi_logger v0.25.3 (*)
└── hikvision_sdk v0.1.0 (/home/ubuntu/Rust/hikvision_sdk)

syn v1.0.109
├── pyo3-macros v0.18.3 (proc-macro)
│   └── pyo3 v0.18.3 (*)
└── pyo3-macros-backend v0.18.3
    └── pyo3-macros v0.18.3 (proc-macro) (*)

syn v2.0.15
├── bindgen v0.65.1 (*)
├── prettyplease v0.2.4
│   └── bindgen v0.65.1 (*)
└── thiserror-impl v1.0.40 (proc-macro)
    └── thiserror v1.0.40
        └── flexi_logger v0.25.3 (*)

**Comment by twistedfall** [comment]
Hi, thanks for the report, do you mind checking the version 0.78 and 0.79 too? It would be really useful to know where the regression was introduced

**Comment by soloist-v** [comment]
`opencv = "0.78.0"`  

error[E0119]: conflicting implementations of trait `std::cmp::Eq` for type `hub::videoio::unnamed`
   --> /home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c7ab931490b4703d/out/opencv/videoio.rs:840:42
    |
459 |     #[derive(Copy, Clone, Debug, PartialEq, Eq)]
    |                                             -- first implementation here
...
840 |     #[derive(Copy, Clone, Debug, PartialEq, Eq)]
    |                                             ^^ conflicting implementation for `hub::videoio::unnamed`
    |
    = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0119, E0428.
For more information about an error, try `rustc --explain E0119`.
error: could not compile `opencv` due to 540 previous errors

Caused by:
  process didn't exit successfully: `CARGO=/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=opencv CARGO_MANIFEST_DIR=/home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.78.2 CARGO_PKG_AUTHORS='Pro <twisted.fall@gmail.com>:Mathieu Poumeyrol <kali@zoy.org>' CARGO_PKG_DESCRIPTION='Rust bindings for OpenCV' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=opencv CARGO_PKG_REPOSITORY='https://github.com/twistedfall/opencv-rust' CARGO_PKG_RUST_VERSION='' CARGO_PKG_VERSION=0.78.2 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=78 CARGO_PKG_VERSION_PATCH=2 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/home/ubuntu/Rust/hikvision_sdk/target/debug/deps:/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib:/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib' OUT_DIR=/home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c7ab931490b4703d/out rustc --crate-name opencv --edition=2021 /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.78.2/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=147 --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="alphamat"' --cfg 'feature="aruco"' --cfg 'feature="aruco_detector"' --cfg 'feature="barcode"' --cfg 'feature="bgsegm"' --cfg 'feature="bioinspired"' --cfg 'feature="calib3d"' --cfg 'feature="ccalib"' --cfg 'feature="cudaarithm"' --cfg 'feature="cudabgsegm"' --cfg 'feature="cudacodec"' --cfg 'feature="cudafeatures2d"' --cfg 'feature="cudafilters"' --cfg 'feature="cudaimgproc"' --cfg 'feature="cudaobjdetect"' --cfg 'feature="cudaoptflow"' --cfg 'feature="cudastereo"' --cfg 'feature="cudawarping"' --cfg 'feature="cvv"' --cfg 'feature="default"' --cfg 'feature="dnn"' --cfg 'feature="dnn_superres"' --cfg 'feature="dpm"' --cfg 'feature="face"' --cfg 'feature="features2d"' --cfg 'feature="flann"' --cfg 'feature="freetype"' --cfg 'feature="fuzzy"' --cfg 'feature="gapi"' --cfg 'feature="hdf"' --cfg 'feature="hfs"' --cfg 'feature="highgui"' --cfg 'feature="img_hash"' --cfg 'feature="imgcodecs"' --cfg 'feature="imgproc"' --cfg 'feature="intensity_transform"' --cfg 'feature="line_descriptor"' --cfg 'feature="mcc"' --cfg 'feature="ml"' --cfg 'feature="objdetect"' --cfg 'feature="optflow"' --cfg 'feature="ovis"' --cfg 'feature="phase_unwrapping"' --cfg 'feature="photo"' --cfg 'feature="plot"' --cfg 'feature="quality"' --cfg 'feature="rapid"' --cfg 'feature="rgbd"' --cfg 'feature="saliency"' --cfg 'feature="sfm"' --cfg 'feature="shape"' --cfg 'feature="stereo"' --cfg 'feature="stitching"' --cfg 'feature="structured_light"' --cfg 'feature="superres"' --cfg 'feature="surface_matching"' --cfg 'feature="text"' --cfg 'feature="tracking"' --cfg 'feature="video"' --cfg 'feature="videoio"' --cfg 'feature="videostab"' --cfg 'feature="viz"' --cfg 'feature="wechat_qrcode"' --cfg 'feature="xfeatures2d"' --cfg 'feature="ximgproc"' --cfg 'feature="xobjdetect"' --cfg 'feature="xphoto"' -C metadata=5ded9b4309e45eb3 -C extra-filename=-5ded9b4309e45eb3 --out-dir /home/ubuntu/Rust/hikvision_sdk/target/debug/deps -L dependency=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps --extern libc=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/liblibc-9fba6e94e0c98d18.rmeta --extern num_traits=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/libnum_traits-04d49bac869cb071.rmeta --extern once_cell=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/libonce_cell-a6df48fd9f0cab30.rmeta --cap-lints warn -L native=/home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c7ab931490b4703d/out -L /usr/lib/x86_64-linux-gnu -l static=ocvrs -l stdc++ -l opencv_shape -l opencv_stitching -l opencv_superres -l opencv_videostab -l opencv_aruco -l opencv_bgsegm -l opencv_bioinspired -l opencv_ccalib -l opencv_datasets -l opencv_dpm -l opencv_face -l opencv_freetype -l opencv_fuzzy -l opencv_hdf -l opencv_line_descriptor -l opencv_optflow -l opencv_video -l opencv_plot -l opencv_reg -l opencv_saliency -l opencv_stereo -l opencv_structured_light -l opencv_phase_unwrapping -l opencv_rgbd -l opencv_viz -l opencv_surface_matching -l opencv_text -l opencv_ximgproc -l opencv_calib3d -l opencv_features2d -l opencv_flann -l opencv_xobjdetect -l opencv_objdetect -l opencv_ml -l opencv_xphoto -l opencv_highgui -l opencv_videoio -l opencv_imgcodecs -l opencv_photo -l opencv_imgproc -l opencv_core --cfg ocvrs_opencv_branch_32 --cfg ocvrs_has_module_aruco --cfg ocvrs_has_module_bgsegm --cfg ocvrs_has_module_bioinspired --cfg ocvrs_has_module_calib3d --cfg ocvrs_has_module_ccalib --cfg ocvrs_has_module_core --cfg ocvrs_has_module_dpm --cfg ocvrs_has_module_face --cfg ocvrs_has_module_features2d --cfg ocvrs_has_module_flann --cfg ocvrs_has_module_freetype --cfg ocvrs_has_module_fuzzy --cfg ocvrs_has_module_hdf --cfg ocvrs_has_module_highgui --cfg ocvrs_has_module_imgcodecs --cfg ocvrs_has_module_imgproc --cfg ocvrs_has_module_line_descriptor --cfg ocvrs_has_module_ml --cfg ocvrs_has_module_objdetect --cfg ocvrs_has_module_optflow --cfg ocvrs_has_module_phase_unwrapping --cfg ocvrs_has_module_photo --cfg ocvrs_has_module_plot --cfg ocvrs_has_module_rgbd --cfg ocvrs_has_module_saliency --cfg ocvrs_has_module_shape --cfg ocvrs_has_module_stereo --cfg ocvrs_has_module_stitching --cfg ocvrs_has_module_structured_light --cfg ocvrs_has_module_superres --cfg ocvrs_has_module_surface_matching --cfg ocvrs_has_module_text --cfg ocvrs_has_module_video --cfg ocvrs_has_module_videoio --cfg ocvrs_has_module_videostab --cfg ocvrs_has_module_viz --cfg ocvrs_has_module_ximgproc --cfg ocvrs_has_module_xobjdetect --cfg ocvrs_has_module_xphoto` (exit status: 1)

**Comment by soloist-v** [comment]
`/home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c7ab931490b4703d/out/opencv/videoio.rs`
```rust

	opencv_type_enum! { crate::videoio::VideoCaptureProperties }
	
	#[repr(C)]
	#[derive(Copy, Clone, Debug, PartialEq, Eq)]
	pub enum VideoWriterProperties {
		VIDEOWRITER_PROP_QUALITY = 1,
		VIDEOWRITER_PROP_FRAMEBYTES = 2,
		VIDEOWRITER_PROP_NSTRIPES = 3,
	}
	
	opencv_type_enum! { crate::videoio::VideoWriterProperties }
	
	#[repr(C)]
	#[derive(Copy, Clone, Debug, PartialEq, Eq)]
	pub enum unnamed {
		CAP_PROP_GSTREAMER_QUEUE_LENGTH = 200,
	}
	
	opencv_type_enum! { crate::videoio::unnamed }
	
	#[repr(C)]
	#[derive(Copy, Clone, Debug, PartialEq, Eq)]
	pub enum unnamed {
		CAP_OPENNI_DEPTH_GENERATOR = -2147483648,
		CAP_OPENNI_IMAGE_GENERATOR = 1073741824,
		CAP_OPENNI_IR_GENERATOR = 536870912,
		CAP_OPENNI_GENERATORS_MASK = -536870912,
	}
	
	opencv_type_enum! { crate::videoio::unnamed }
	
	#[repr(C)]
	#[derive(Copy, Clone, Debug, PartialEq, Eq)]
	pub enum unnamed {
		CAP_PROP_OPENNI_OUTPUT_MODE = 100,
		CAP_PROP_OPENNI_FRAME_MAX_DEPTH = 101,
		CAP_PROP_OPENNI_BASELINE = 102,
		CAP_PROP_OPENNI_FOCAL_LENGTH = 103,
		CAP_PROP_OPENNI_REGISTRATION = 104,
		// Duplicate, use CAP_PROP_OPENNI_REGISTRATION instead
		// CAP_PROP_OPENNI_REGISTRATION_ON = 104,
		CAP_PROP_OPENNI_APPROX_FRAME_SYNC = 105,
		CAP_PROP_OPENNI_MAX_BUFFER_SIZE = 106,
		CAP_PROP_OPENNI_CIRCLE_BUFFER = 107,
		CAP_PROP_OPENNI_MAX_TIME_DURATION = 108,
		CAP_PROP_OPENNI_GENERATOR_PRESENT = 109,
		CAP_PROP_OPENNI2_SYNC = 110,
		CAP_PROP_OPENNI2_MIRROR = 111,
	}
	
	opencv_type_enum! { crate::videoio::unnamed }
	
	#[repr(C)]
	#[derive(Copy, Clone, Debug, PartialEq, Eq)]
	pub enum unnamed {
		CAP_OPENNI_IMAGE_GENERATOR_PRESENT = 1073741933,
		CAP_OPENNI_IMAGE_GENERATOR_OUTPUT_MODE = 1073741924,
		CAP_OPENNI_DEPTH_GENERATOR_PRESENT = -2147483539,
		CAP_OPENNI_DEPTH_GENERATOR_BASELINE = -2147483546,
		CAP_OPENNI_DEPTH_GENERATOR_FOCAL_LENGTH = -2147483545,
		CAP_OPENNI_DEPTH_GENERATOR_REGISTRATION = -2147483544,
		// Duplicate, use CAP_OPENNI_DEPTH_GENERATOR_REGISTRATION instead
		// CAP_OPENNI_DEPTH_GENERATOR_REGISTRATION_ON = -2147483544,
		CAP_OPENNI_IR_GENERATOR_PRESENT = 536871021,
	}
	
	opencv_type_enum! { crate::videoio::unnamed }
```

**Comment by soloist-v** [comment]
`opencv = "0.79.0"`
error[E0428]: the name `unnamed` is defined multiple times
   --> /home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c9fe25b002269b4c/out/opencv/calib3d.rs:300:2
    |
287 |     pub enum unnamed {
    |     ---------------- previous definition of the type `unnamed` here
...
300 |     pub enum unnamed {
    |     ^^^^^^^^^^^^^^^^ `unnamed` redefined here
    |
    = note: `unnamed` must be defined only once in the type namespace of this module

error[E0428]: the name `unnamed` is defined multiple times
   --> /home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c9fe25b002269b4c/out/opencv/calib3d.rs:316:2
    |
287 |     pub enum unnamed {
    |     ---------------- previous definition of the type `unnamed` here
...
316 |     pub enum unnamed {
    |     ^^^^^^^^^^^^^^^^ `unnamed` redefined here
    |
    = note: `unnamed` must be defined only once in the type namespace of this module

error[E0428]: the name `unnamed` is defined multiple times
   --> /home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c9fe25b002269b4c/out/opencv/calib3d.rs:327:2

error[E0119]: conflicting implementations of trait `std::cmp::Eq` for type `hub::videoio::unnamed`
   --> /home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c9fe25b002269b4c/out/opencv/videoio.rs:823:42
    |
459 |     #[derive(Copy, Clone, Debug, PartialEq, Eq)]
    |                                             -- first implementation here
...
823 |     #[derive(Copy, Clone, Debug, PartialEq, Eq)]
    |                                             ^^ conflicting implementation for `hub::videoio::unnamed`
    |
    = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0119]: conflicting implementations of trait `std::cmp::Eq` for type `hub::videoio::unnamed`
   --> /home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c9fe25b002269b4c/out/opencv/videoio.rs:840:42
    |
459 |     #[derive(Copy, Clone, Debug, PartialEq, Eq)]
    |                                             -- first implementation here
...
840 |     #[derive(Copy, Clone, Debug, PartialEq, Eq)]
    |                                             ^^ conflicting implementation for `hub::videoio::unnamed`
    |
    = note: this error originates in the derive macro `Eq` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0119, E0428.
For more information about an error, try `rustc --explain E0119`.
error: could not compile `opencv` due to 540 previous errors

Caused by:
  process didn't exit successfully: `CARGO=/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/bin/cargo CARGO_CRATE_NAME=opencv CARGO_MANIFEST_DIR=/home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.79.0 CARGO_PKG_AUTHORS='Pro <twisted.fall@gmail.com>:Mathieu Poumeyrol <kali@zoy.org>' CARGO_PKG_DESCRIPTION='Rust bindings for OpenCV' CARGO_PKG_HOMEPAGE='' CARGO_PKG_LICENSE=MIT CARGO_PKG_LICENSE_FILE='' CARGO_PKG_NAME=opencv CARGO_PKG_REPOSITORY='https://github.com/twistedfall/opencv-rust' CARGO_PKG_RUST_VERSION=1.59 CARGO_PKG_VERSION=0.79.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=79 CARGO_PKG_VERSION_PATCH=0 CARGO_PKG_VERSION_PRE='' LD_LIBRARY_PATH='/home/ubuntu/Rust/hikvision_sdk/target/debug/deps:/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib:/home/ubuntu/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib' OUT_DIR=/home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c9fe25b002269b4c/out rustc --crate-name opencv --edition=2021 /home/ubuntu/.cargo/registry/src/github.com-1ecc6299db9ec823/opencv-0.79.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=147 --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="alphamat"' --cfg 'feature="aruco"' --cfg 'feature="aruco_detector"' --cfg 'feature="barcode"' --cfg 'feature="bgsegm"' --cfg 'feature="bioinspired"' --cfg 'feature="calib3d"' --cfg 'feature="ccalib"' --cfg 'feature="cudaarithm"' --cfg 'feature="cudabgsegm"' --cfg 'feature="cudacodec"' --cfg 'feature="cudafeatures2d"' --cfg 'feature="cudafilters"' --cfg 'feature="cudaimgproc"' --cfg 'feature="cudaobjdetect"' --cfg 'feature="cudaoptflow"' --cfg 'feature="cudastereo"' --cfg 'feature="cudawarping"' --cfg 'feature="cvv"' --cfg 'feature="default"' --cfg 'feature="dnn"' --cfg 'feature="dnn_superres"' --cfg 'feature="dpm"' --cfg 'feature="face"' --cfg 'feature="features2d"' --cfg 'feature="flann"' --cfg 'feature="freetype"' --cfg 'feature="fuzzy"' --cfg 'feature="gapi"' --cfg 'feature="hdf"' --cfg 'feature="hfs"' --cfg 'feature="highgui"' --cfg 'feature="img_hash"' --cfg 'feature="imgcodecs"' --cfg 'feature="imgproc"' --cfg 'feature="intensity_transform"' --cfg 'feature="line_descriptor"' --cfg 'feature="mcc"' --cfg 'feature="ml"' --cfg 'feature="objdetect"' --cfg 'feature="optflow"' --cfg 'feature="ovis"' --cfg 'feature="phase_unwrapping"' --cfg 'feature="photo"' --cfg 'feature="plot"' --cfg 'feature="quality"' --cfg 'feature="rapid"' --cfg 'feature="rgbd"' --cfg 'feature="saliency"' --cfg 'feature="sfm"' --cfg 'feature="shape"' --cfg 'feature="stereo"' --cfg 'feature="stitching"' --cfg 'feature="structured_light"' --cfg 'feature="superres"' --cfg 'feature="surface_matching"' --cfg 'feature="text"' --cfg 'feature="tracking"' --cfg 'feature="video"' --cfg 'feature="videoio"' --cfg 'feature="videostab"' --cfg 'feature="viz"' --cfg 'feature="wechat_qrcode"' --cfg 'feature="xfeatures2d"' --cfg 'feature="ximgproc"' --cfg 'feature="xobjdetect"' --cfg 'feature="xphoto"' -C metadata=9442348f77fe0d93 -C extra-filename=-9442348f77fe0d93 --out-dir /home/ubuntu/Rust/hikvision_sdk/target/debug/deps -L dependency=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps --extern libc=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/liblibc-9fba6e94e0c98d18.rmeta --extern num_traits=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/libnum_traits-04d49bac869cb071.rmeta --extern once_cell=/home/ubuntu/Rust/hikvision_sdk/target/debug/deps/libonce_cell-a6df48fd9f0cab30.rmeta --cap-lints warn -L native=/home/ubuntu/Rust/hikvision_sdk/target/debug/build/opencv-c9fe25b002269b4c/out -L /usr/lib/x86_64-linux-gnu -l static=ocvrs -l stdc++ -l opencv_shape -l opencv_stitching -l opencv_superres -l opencv_videostab -l opencv_aruco -l opencv_bgsegm -l opencv_bioinspired -l opencv_ccalib -l opencv_datasets -l opencv_dpm -l opencv_face -l opencv_freetype -l opencv_fuzzy -l opencv_hdf -l opencv_line_descriptor -l opencv_optflow -l opencv_video -l opencv_plot -l opencv_reg -l opencv_saliency -l opencv_stereo -l opencv_structured_light -l opencv_phase_unwrapping -l opencv_rgbd -l opencv_viz -l opencv_surface_matching -l opencv_text -l opencv_ximgproc -l opencv_calib3d -l opencv_features2d -l opencv_flann -l opencv_xobjdetect -l opencv_objdetect -l opencv_ml -l opencv_xphoto -l opencv_highgui -l opencv_videoio -l opencv_imgcodecs -l opencv_photo -l opencv_imgproc -l opencv_core --cfg ocvrs_opencv_branch_32 --cfg ocvrs_has_module_aruco --cfg ocvrs_has_module_bgsegm --cfg ocvrs_has_module_bioinspired --cfg ocvrs_has_module_calib3d --cfg ocvrs_has_module_ccalib --cfg ocvrs_has_module_core --cfg ocvrs_has_module_dpm --cfg ocvrs_has_module_face --cfg ocvrs_has_module_features2d --cfg ocvrs_has_module_flann --cfg ocvrs_has_module_freetype --cfg ocvrs_has_module_fuzzy --cfg ocvrs_has_module_hdf --cfg ocvrs_has_module_highgui --cfg ocvrs_has_module_imgcodecs --cfg ocvrs_has_module_imgproc --cfg ocvrs_has_module_line_descriptor --cfg ocvrs_has_module_ml --cfg ocvrs_has_module_objdetect --cfg ocvrs_has_module_optflow --cfg ocvrs_has_module_phase_unwrapping --cfg ocvrs_has_module_photo --cfg ocvrs_has_module_plot --cfg ocvrs_has_module_rgbd --cfg ocvrs_has_module_saliency --cfg ocvrs_has_module_shape --cfg ocvrs_has_module_stereo --cfg ocvrs_has_module_stitching --cfg ocvrs_has_module_structured_light --cfg ocvrs_has_module_superres --cfg ocvrs_has_module_surface_matching --cfg ocvrs_has_module_text --cfg ocvrs_has_module_video --cfg ocvrs_has_module_videoio --cfg ocvrs_has_module_videostab --cfg ocvrs_has_module_viz --cfg ocvrs_has_module_ximgproc --cfg ocvrs_has_module_xobjdetect --cfg ocvrs_has_module_xphoto` (exit status: 1)

**Comment by twistedfall** [comment]
Can you please do a `cargo clean` and then `cargo build -vv` and post the full log produced by the last command? Please use crate version 0.80.0.

**Comment by soloist-v** [comment]
https://gist.github.com/soloist-v/e8d1fe0c7cfe4a3f812b3ae5a0ea86d2

**Comment by twistedfall** [comment]
Ah, I see now, you're using OpenCV 3.2.0, this is not supported anymore, please switch to 3.4.x or 4.x.

**Comment by soloist-v** [comment]
When I use OpenCV 4.5.5, I also encounter the same issue.
https://gist.github.com/soloist-v/7d4f7093911761b12f85119acd1f3e04

**Comment by twistedfall** [comment]
Thanks for checking! I'll investigate

**Comment by jaques-sam** [comment]
Encountering the same on in Debian 10 (docker: lovebug356/rust-buster)

**Comment by twistedfall** [comment]
This should be fixed in v0.81.0, it was because of the quirks in the old clang-6.
